### PR TITLE
boards/cxd56xx/spresense: add fs automount driver for SD Card

### PR DESCRIPTION
**Pull request by pkarashchenko**
## Summary
Add FS automount driver for SD Card.

This is the first step to remove `BOARDIOC_SDCARD_SETNOTIFYCB`

Options to enable functionality:
```
CONFIG_CXD56_SDCARD_AUTOMOUNT=y
CONFIG_FS_AUTOMOUNTER=y
CONFIG_FS_AUTOMOUNTER_DRIVER=y
```

## Impact
New `nsh_automount` configuration added

## Testing
Tested with spresense board

### CODE REVIEW

**Comment by pkarashchenko** [comment]
@SPRESENSE have you had a chance to try this changes?

**Comment by pkarashchenko** [comment]
@SPRESENSE maybe you can recommend me the board (send me link to the board), so I can purchase it locally and try changes by myself?

**Comment by xiaoxiang781216** [comment]
@jerpelea @masayuki2009  could you help verify the change?

**Comment by jerpelea** [comment]
@xiaoxiang781216 I cant test until next week 
@masayuki2009 can you please test the code?

**Comment by jerpelea** [comment]
> @SPRESENSE maybe you can recommend me the board (send me link to the board), so I can purchase it locally and try changes by myself?

@pkarashchenko please contact me by mail

**Comment by masayuki2009** [comment]
@pkarashchenko 

```
CC:  board/cxd56_sdcard.c
board/cxd56_sdcard.c: In function 'board_sdcard_disable':
board/cxd56_sdcard.c:205:7: warning: unused variable 'ret' [-Wunused-variable]
  205 |   int ret;
      |       ^~~
```

**Comment by masayuki2009** [comment]
@pkarashchenko 

I added the following configs to `spresense:rndis`

```
CONFIG_CXD56_SDCARD_AUTOMOUNT=y
CONFIG_FS_AUTOMOUNTER=y
CONFIG_FS_AUTOMOUNTER_DRIVER=y
```

However, the sd card was not mounted.

```
NuttShell (NSH) NuttX-3.6.1
nsh> uname -a
NuttX  3.6.1 f590b13f0a-dirty Dec  7 2022 09:36:05 arm spresense
nsh> ps
  PID GROUP PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0 FIFO     Kthread N-- Ready              00000000 001000 000384  38.4%  Idle Task
    1     1 224 RR       Kthread --- Waiting  Semaphore 00000000 001992 000300  15.0%  hpwork 0x2d053ad0
    2     2 100 RR       Kthread --- Waiting  Semaphore 00000000 001992 000300  15.0%  lpwork 0x2d053ae8
    3     3 100 RR       Task    --- Running            00000000 003024 001444  47.7%  spresense_main
    4     4 200 RR       Task    --- Waiting  MQ empty  00000000 000976 000456  46.7%  cxd56_pm_task
    6     6 100 RR       Task    --- Waiting  Semaphore 00000000 002008 000620  30.8%  telnetd
nsh> mount
  /mnt/spif type smartfs
  /proc type procfs
```

**Comment by pkarashchenko** [comment]
@masayuki2009 thank you very much for spending time on this. So I will need to wait some time until I get the board and will debug it "on the desk".

**Comment by pkarashchenko** [comment]
@masayuki2009 @jerpelea finally I was able to finalize this PR and test with spresense board. Could you please try `spresense:nsh_automount` config at your end?

File: boards/arm/cxd56xx/spresense/src/cxd56_sdcard.c [hunk 1]
```diff
@@ -351,14 +349,30 @@ int board_sdcard_initialize(void)
...
+  /* Enable SDC if card is inserted */
+
+  if (g_sdhci.inserted)
+    {
+      /* Card Detect = Present, Write Protect = disable */
+
+      putreg32(0, CXD56_TOPREG_IOFIX_APP);
+
+      board_sdcard_enable(NULL);
+    }
```
Comment (pkarashchenko) [inline boards/arm/cxd56xx/spresense/src/cxd56_sdcard.c:369]: Maybe it is better just to call `board_sdcard_detect_int` instead.

**Comment by pkarashchenko** [comment]
@masayuki2009 instead of adding a new config I can add SD card automount to all existing configs with `CXD56_SDIO=y`. @masayuki2009 what do you think?

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@pkarashchenko 

>@masayuki2009 @jerpelea finally I was able to finalize this PR and test with spresense board. Could you please try spresense:nsh_automount config at your end?

I confirmed that `spresense:nsh_automount` works with my board.

>@masayuki2009 instead of adding a new config I can add SD card automount to all existing configs with CXD56_SDIO=y. What do you think? But now it is either FS automount or a driver callback, but not both. So maybe we need to make changes in few steps.

I think that adding a new config for automount is enough so far.

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**